lib/babe/verify.go

```diff
@@ -47,11 +49,25 @@ type VerificationManager struct {
 
 // NewVerificationManager returns a new NewVerificationManager
 func NewVerificationManager(blockState BlockState, epochState EpochState) (*VerificationManager, error) {
-	if blockState == nil {
+	var isNilBlock bool
+	switch blockState.(type) {
+	case *state.BlockState:
+		isNilBlock = blockState == (*state.BlockState)(nil)
+	default:
+		isNilBlock = reflect.ValueOf(blockState).IsNil()
+	}
+	if isNilBlock {
 		return nil, errNilBlockState
 	}
 
-	if epochState == nil {
+	var isNilEpoch bool
+	switch epochState.(type) {
+	case *state.EpochState:
+		isNilEpoch = epochState == (*state.EpochState)(nil)
+	default:
+		isNilEpoch = reflect.ValueOf(epochState).IsNil()
+	}
+	if isNilEpoch {
```

Comment:
do we ned this `isNilBlock = blockState == (*state.BlockState)(nil)` or does `reflect.ValueOf(blockState).IsNil()` evaluate to the same thing? I'd rather not have the `(*state.BlockState)` type assertion if possible